1.19 blog post (#6486)
prettier/prettier: website/blog/2019-09-25-1.19.0.md

---
author: Lucas Duailibe (@duailibe)
authorURL: https://github.com/duailibe
title: "Prettier 1.19: Vue script and style indentation, and skipping unchanged files"
---

> This release includes as asual lots of fixes and improvements, but most importantly includes a feature very requested by the Vue community to indent `script` and `style` tags contents and a flag `--only-changed` to speed up Prettier CLI runs.

Reviewer: misspelling
```suggestion
This release includes as usual lots of fixes and improvements, but most importantly includes a feature much requested by the Vue community to indent `script` and `style` tags' contents and a flag `--only-changed` to speed up Prettier CLI runs.
```

<!-- prettier-ignore -->
```js
// Input
foo(
  ({
    a,

    b
  }) => {}
);

// Output (Prettier 1.18)
foo(({ a,
  b }) => {});

// Output (Prettier 1.19)
foo(
  ({
    a,

    b
  }) => {}
);
```

#### Keep unary expressions parentheses with comments ([#6217] by [@sosukesuzuki])

Previously, Prettier removes parentheses enclose unary expressions. This change modify to keep it when the expression has comments.

<!-- prettier-ignore -->
```ts
// Input
!(
  /* foo */
  foo
);
!(
  foo // foo
);

// Output (Prettier 1.18)
!/* foo */
foo;
!foo; // foo

// Output (Prettier 1.19)
!(/* foo */ foo);
!(
  foo // foo
);
```

#### Stop moving comments inside tagged template literals ([#6236] by [@sosukesuzuki])

Previously, Prettier would move comments after the tag inside the template literal. This version fixes this problem.

<!-- prettier-ignore -->
```js
// Input
foo //comment
`
`;

// Output (Prettier 1.18)
foo` // comment
`;

// Output (Prettier 1.19)
foo // comment
`
`;
```

#### Fix moving comments in function calls like `useEffect` second argument ([#6270] by [@sosukesuzuki])

> This fixes a bug that was affecting function calls that have a arrow function as first argument and an array expression as second argument, such as the common React's `useEffect`. A comment in its own line before the second argument would be moved to the line above.

Reviewer: ```suggestion
This fixes a bug that was affecting function calls that have an arrow function as first argument and an array expression as second argument, such as the common React's `useEffect`. A comment in its own line before the second argument would be moved to the line above.
```

<p>
  Hi
  {{firstName}}
  {{lastName}}
</p>

// Output (Prettier 1.19)
<p>
  Your username is @{{name}}
</p>
<p>
  Hi {{firstName}} {{lastName}}
</p>
```

#### Improve comment formatting ([#6206] by [@gavinjoyce])

Previously, Prettier would sometimes ignore whitespace when formatting comments.

<!-- prettier-ignore -->
```hbs
// Input
<div>
  {{! Foo }}
  {{#if @foo}}
    Foo
  {{/if}}

  {{! Bar }}
  {{#if @bar}}
    Bar
  {{/if}}
</div>

// Output (Prettier 1.18)
<div>
  {{! Foo }}
  {{#if @foo}}
    Foo
  {{/if}}{{! Bar }}{{#if @bar}}
    Bar
  {{/if}}
</div>

// Output (Prettier 1.19)
<div>
  {{! Foo }}
  {{#if @foo}}
    Foo
  {{/if}}
  {{! Bar }}
  {{#if @bar}}
    Bar
  {{/if}}
</div>
```

#### Improve comment formatting ([#6234] by [@gavinjoyce])

Previously, Prettier would incorrectly decode HTML entiites.

<!-- prettier-ignore -->
```hbs
// Input
<p>
  Some escaped characters: &lt; &gt; &amp;
</p>

// Output (Prettier 1.18)
<p>
  Some escaped characters: < > &
</p>

// Output (Prettier 1.19)
<p>
  Some escaped characters: &lt; &gt; &amp;
</p>
```

### Vue

#### Don't break the template element included in a line shorter than print width ([#6284] by [@sosukesuzuki])

> Previously, even if the line length is shorter than print-width is Prettier breaks the line with a template element.

Reviewer: ```suggestion
Previously, even if the line length is shorter than print-width, Prettier broke the line with a template element.
```
